tests/_pyodide/test_pyodide_session.py

```diff
@@ -92,44 +92,27 @@ def mock_pyodide() -> Generator[MagicMock, None, None]:
         del sys.modules["pyodide"]
 
 
-@pytest.fixture
-def app_metadata(app_file: Path) -> AppMetadata:
-    return AppMetadata(
-        query_params={},
-        cli_args={},
-        app_config=_AppConfig(),
-        filename=str(app_file),
-    )
-
-
-@pytest.fixture
-def app_file_manager(app_file: Path) -> AppFileManager:
-    return AppFileManager(filename=str(app_file))
-
-
-@pytest.fixture
-def async_queue_manager() -> AsyncQueueManager:
-    return AsyncQueueManager()
-
-
 @pytest.fixture
 def pyodide_session(
-    app_file_manager: AppFileManager,
     mock_on_write: Callable[[KernelMessage], None],
-    app_metadata: AppMetadata,
+    pyodide_app_file: Path,
 ) -> PyodideSession:
     return PyodideSession(
-        app=app_file_manager,
+        app=AppFileManager(filename=str(pyodide_app_file)),
```

Comment:
Oh tricky, may worth leaving a comment somewhere? I guess we both know now, but definitely not an immediately obvious one